conans/client/export.py

```diff
@@ -25,12 +25,31 @@ def export_conanfile(output, paths, file_patterns, origin_folder, conan_ref, kee
     if previous_digest and previous_digest.file_sums == digest.file_sums:
         digest = previous_digest
         output.info("The stored package has not changed")
+        modified_recipe = False
     else:
         output.success('A new %s version was exported' % CONANFILE)
-        if not keep_source:
-            rmdir(paths.source(conan_ref))
-        output.success('%s exported to local storage' % CONANFILE)
-        output.success('Folder: %s' % destination_folder)
+        output.info('Folder: %s' % destination_folder)
+        modified_recipe = True
+
+    source = paths.source(conan_ref)
+    dirty = os.path.join(source, DIRTY_FILE)
+    remove = False
+    if os.path.exists(dirty):
+        output.info("Source folder is dirty, forcing removal")
+        remove = True
+    elif modified_recipe and not keep_source and os.path.exists(source):
+        output.info("Package recipe modified in export, forcing source folder removal")
+        output.info("Use the --keep-source, -k option to skip it")
+        remove = True
+    if remove:
+        output.info("Removing 'source' folder, this can take a while for big packages")
+        try:
+            rmdir(source)
+        except Exception as e:
```

Comment:
I guess no, KeyboardInterrupt does not inherit from Exception but BaseException. Changing it.